### DISCUSSION TITLE
Data Structure: Ticker Identifier

### DISCUSSION THREAD

**Discussion by mkitzmann** [GoPlan-Finance/goplan-app#33] [Q&A]
Every asset would have to have a unique identifier. For most financial assets this would be the ticker or symbol (eg. `AAPL`). Do we want to use the ticker, an autoincremented id or a UUID as identifier?

Since we will eventually be getting price information from different APIs, we will somehow have to map the different ticker/symbols to our identifier. This would probably result in a large dataset which we would have to update on a regular basis, so it doesnt get outdated. How could we achieve this?

**Comment by sadortun** [comment]
FYI, i'll try to put a few things in place so you get a better view on how's Parse works. but in short, everything have a unique ID which is a 8-10 characher string.

I'll also prepare a [Google Sheet with a proposed DB schema ](https://docs.google.com/spreadsheets/d/1M44J8ULjC1JY7WpC6xt-O3_C2u1394eYrsxpdhvo-q8/edit#gid=0)~~later today or tomorow.~~  

You can also create relations and pointers https://docs.parseplatform.org/js/guide/#relational-data

```js
const post = new Post();
 // post.id = "1zEcyElZ80";

myComment.set("parent", post);
```

Since we can get the same ticker on different exchanges, and various APIs will provide this data differently we will most likely need to have :


- StockTicker
  -  name
  - parentExchange
UNIQUE INDEX (  name + parentExchange )   

- StockExchange
   - name
UNIQUE INDEX(  name ) 
   
- StockProviderAdapter
   - name
